docs/design/features/source-generator-pinvokes.md

```diff
@@ -0,0 +1,224 @@
+
+        /// <summary>
+        /// Indicates how to marshal string parameters to the method and controls name mangling.
+        /// </summary>
+        /// <see cref="System.Runtime.InteropServices.DllImportAttribute.CharSet"/>
+        public CharSet CharSet;
+
+        /// <summary>
+        /// Indicates the name or ordinal of the DLL entry point to be called.
+        /// </summary>
+        /// <see cref="System.Runtime.InteropServices.DllImportAttribute.EntryPoint"/>
+        public string? EntryPoint;
+
+        /// <summary>
+        /// Controls whether the System.Runtime.InteropServices.DllImportAttribute.CharSet
+        /// field causes the common language runtime to search an unmanaged DLL for entry-point
+        /// names other than the one specified.
+        /// </summary>
+        /// <see cref="System.Runtime.InteropServices.DllImportAttribute.ExactSpelling"/>
+        public bool ExactSpelling;
+
+        /// <summary>
+        /// Indicates whether unmanaged methods that have HRESULT or retval return values
+        /// are directly translated or whether HRESULT or retval return values are automatically
+        /// converted to exceptions.
+        /// </summary>
+        /// <see cref="System.Runtime.InteropServices.DllImportAttribute.PreserveSig"/>
+        public bool PreserveSig;
+
+        /// <summary>
+        /// Indicates whether the callee calls the SetLastError Windows API function before
+        /// returning from the attributed method.
+        /// </summary>
+        /// <see cref="System.Runtime.InteropServices.DllImportAttribute.SetLastError"/>
+        public bool SetLastError;
+
+        /// <summary>
+        /// Enables or disables the throwing of an exception on an unmappable Unicode character
+        /// that is converted to an ANSI "?" character.
+        /// </summary>
+        /// <see cref="System.Runtime.InteropServices.DllImportAttribute.ThrowOnUnmappableChar"/>
+        public bool ThrowOnUnmappableChar;
+    }
+}
+```
+
+## FAQs
+
+* Can the above API be used to provide a reverse P/Invoke stub?
+
+    * No. Reverse P/Invoke invocation is performed via a delegate and integrating this proposal would prove difficult. Alternative approach is to leverage the [`NativeCallableAttribute`](https://github.com/dotnet/runtime/issues/32462) feature.
+
+* How will users get error messages during source generator?
+
+    * The Source Generator API will be permitted to provide warnings and errors through the [Roslyn SDK](https://docs.microsoft.com/dotnet/csharp/roslyn-sdk/).
+
+* Is it be possible to completely replicate the marshaling rules in the current built-in system using existing .NET APIs?
+
+    * No. There are rules and semantics that would be difficult to replicate with the current .NET API surface. Additional .NET APIs will likely need to be added in order to allow a Source Generator implementation to provide identical semantics with the built-in system.
```

Comment:
[This](https://github.com/dotnet/runtime/pull/33742#discussion_r395382954) example should be mentioned (not everybody reads every comment on PR, especially outdated comments).